pkg/asset/manifests/additionaltrustbundleconfig.go

```diff
@@ -0,0 +1,125 @@
+package manifests
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+)
+
+var (
+	additionalTrustBundleConfigFileName = filepath.Join(manifestDir, "cluster-proxy-ca-bundle-01-config.yaml")
+)
+
+const (
+	additionalTrustBundleConfigDataKey = "config"
+)
+
+// AdditionalTrustBundleConfig generates the additional-trust-bundle-config.yaml files.
+type AdditionalTrustBundleConfig struct {
+	ConfigMap *corev1.ConfigMap
+	File      *asset.File
+}
+
+var _ asset.WritableAsset = (*AdditionalTrustBundleConfig)(nil)
+
+// Name returns a human friendly name for the asset.
+func (*AdditionalTrustBundleConfig) Name() string {
+	return "Additional Trust Bundle Config"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate
+// the asset.
+func (*AdditionalTrustBundleConfig) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+	}
+}
+
+// Generate generates the CloudProviderConfig.
+func (atbc *AdditionalTrustBundleConfig) Generate(dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	dependencies.Get(installConfig)
+
+	if installConfig.Config.AdditionalTrustBundle == "" {
+		return nil
+	}
+	data, err := parseCertificates(installConfig.Config.AdditionalTrustBundle)
+
+	if err != nil {
+		return err
+	}
+
+	cm := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "openshift-config",
+			Name:      "proxy-ca-bundle",
```

Comment:
name is somethings like `user-ca-bundle`

Reply:
@abhinavdahiya that was the name based on the google doc.

Reply:
so looking at
https://github.com/openshift/api/blob/d75a161a0f4ded1815ec24321fc1f8609c255cdb/config/v1/types_proxy.go#L46-L49

namespace is required to be `openshift-config-managed`
the key should be `ca-bundle.crt`

so restriction on name, and since we use this for more than proxy, `user-ca-bundle` like is more appropriate.